### DISCUSSION TITLE
Sonoff Mini Fisical Switch JSON config

### DISCUSSION THREAD

**Discussion by ErikDGC** [RavenSystem/esp-homekit-devices#1665] [MEPLHAA Scripts]
Hello, hope your all having a lovely week. I was able to flash a Sonoff Mini to be compatible with Homekit. The issue is that I have a fisical switch attached to it & I'm not good with JSON config. So I got one off the web but the fisical switch is rendered useless & can only be controled thru homekit. Can anyone please help me with the correct config so that the sonoff mini can be controled thru both homekit & the switch connected to the sonoff as seen in the following YouTube video. [Sonoff homekit video](https://m.youtube.com/watch?v=nrXdOczHWc4  )  If anyone can please provide any details on how to go about this I would really appreciate it., thanks

**Comment by vsidlu** [comment]
Hi!

You can try this one. Working for me

{"c":{"o":3,"l":13,"i":1,"n":"sonoff_mini","b":[{"g":0,"t":5}]},"a":[{"t":2,"s":5,"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":4,"t":1}]}]}

external push button attached

**Comment by Ucoz3000** [comment]
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4},{"g":4,"t":0}],"s":0}]}

**Comment by ErikDGC** [comment]
Thank you both for providing me with the JSON config. Will be trying them out shortly.
